### DISCUSSION TITLE
Windows 2000-style separators between toolbars?

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#379] [Ideas]
I liked this very much and was very disappointed at the time with Win XP that lacked these vertical separators between toolbars:

![изображение](https://user-images.githubusercontent.com/1271736/141869747-afaea687-76a1-4bbf-9946-f8ca42700372.png)

Now, we can add them 
![изображение](https://user-images.githubusercontent.com/1271736/141869821-cd9372b9-eb7a-4ff6-9332-5ce554ca8530.png)
by adding RBS_BANDBOARDERS (0x400) to ReBarWindow32.

It also adds horizontal separators:

![изображение](https://user-images.githubusercontent.com/1271736/141870430-ec5aef3f-ebd8-4610-acf9-0c84ea11bb9c.png)
![изображение](https://user-images.githubusercontent.com/1271736/141870614-5515bc82-60e6-4b98-8d7e-35af42dcba29.png)

The only difference is that the first toolbar does not receive the separator. Still, I think, this would be a nice option.

**Comment by valinet** [comment]
Interesting. Works with non-classic theme as well. A nice addition, I guess. Implemented in the latest build (in "Taskbar" section):

![image](https://user-images.githubusercontent.com/6503598/141878128-5902faab-0b1c-4838-a776-11b92066fe5a.png)

> by adding RBS_BANDBOARDERS (0x400) to ReBarWindow32.

Thanks for specifying this, it is much more easier when you know what you have to do. It easily pointed me in the right direction.

Thank you

**Reply by Anixx** [reply]
Thanks for implementing! I will actually use it!

This also may be interesting:

![изображение](https://user-images.githubusercontent.com/1271736/142438956-7b21a01d-1350-41ca-9836-744fd1d4aed2.png)

Can be achieved by removing TBSTYLE_FLAT (0x800) from ToolbarWindow32. 

This is not, of course, how it was in any previous version of Windows, except Chicago beta:
![изображение](https://user-images.githubusercontent.com/1271736/142439770-d60d78ef-26b5-4bb8-955f-643c3ca4a396.png)
![изображение](https://user-images.githubusercontent.com/1271736/142439841-96185eb8-790e-41a0-8396-572345bb55ef.png)

**Reply by Anixx** [reply]
And, also this:
![изображение](https://user-images.githubusercontent.com/1271736/142476070-8bf0ff2f-851a-4ed8-a712-daa6a2654e5a.png)
The Windows 2000/XP style of the toolbars in file manager.
(topic on WinClassic: https://winclassic.boards.net/thread/244/classic-toolbar-borders-explorer )

Achieved by the following styles applied to ReBarWindow32:

* Add WS_BORDER (0x800000) 
* Add RBS_BANDBORDERS (0x400)
* Remove RBS_VARHEIGHT (0x200) - this one so to make the heights of all toolbars equal.

**Reply by HitAnyKey** [reply]
> 
> 
> And, also this:
> ![изображение](https://user-images.githubusercontent.com/1271736/142476070-8bf0ff2f-851a-4ed8-a712-daa6a2654e5a.png)
>The Windows 2000/XP style of the toolbars in file manager.
> (topic on WinClassic: https://winclassic.boards.net/thread/244/classic-toolbar-borders-explorer )
> 
> Achieved by the following styles applied to ReBarWindow32:
> 
>     * Add WS_BORDER (0x800000)
> 
>     * Add RBS_BANDBORDERS (0x400)
> 
>     * Remove RBS_VARHEIGHT (0x200) - this one so to make the heights of all toolbars equal.

It is useful because once this is implemented, we no longer need a script to remove the navigation bar and a script to add a toolbar border.

**Reply by Anixx** [reply]
Removal of navigation bar is already implemented in ExplorerPatcher.

**Reply by HitAnyKey** [reply]
Yeah, I know it. But I can't add a toolbar border when navigation bar is removed by Explorer Patcher. So I need to remove a navigation bar using AHK script for now.
